src/fiatconnect/saga.ts

```diff
@@ -494,8 +495,19 @@ export function* handleSelectFiatConnectQuote({
       fiatAccountType: quote.getFiatAccountType(),
     })
 
+    // This is expected when the user has not yet created a fiatAccount with the provider
     if (!fiatAccount) {
-      // This is expected when the user has not yet created a fiatAccount with the provider
+      // If the quote has kyc, navigate to the second step of the KycLanding page
+      if (kycSchema) {
+        navigate(Screens.KycLanding, {
+          quote,
+          flow: quote.flow,
+          step: 'two',
+        })
+        yield put(selectFiatConnectQuoteCompleted())
```

Comment:
do we need the yield delay(500) as in the else part? And a minor nit, could also do 
```js
if (kycSchema) {
   navigate(Screens.KycLanding, {..})
} else {
  navigate(Screens.FiatConnectLinkAccount, {..})
  yield delay(500) // can be outside if else if both requires it.
}
yield put(selectFiatConnectQuoteCompleted())
return
```